os-eap-sso/added/keycloak.sh

```diff
@@ -157,17 +158,17 @@ function explode_keycloak_deployments() {
     if [ -f "${JBOSS_HOME}/standalone/deployments/${sso_deployment}/WEB-INF/web.xml" ]; then
       requested_auth_method=`cat ${JBOSS_HOME}/standalone/deployments/${sso_deployment}/WEB-INF/web.xml | xmllint --nowarning --xpath "string(//*[local-name()='auth-method'])" - | sed ':a;N;$!ba;s/\n//g' | tr -d '[:space:]'`
       sed -i "s|${requested_auth_method}|${auth_method}|" "${JBOSS_HOME}/standalone/deployments/${sso_deployment}/WEB-INF/web.xml"
-    fi    
+    fi
   done
 }
 
 function get_token() {
 
   token=""
   if [ -n "$SSO_USERNAME" ] && [ -n "$SSO_PASSWORD" ]; then
-    token=`$CURL --data "username=${SSO_USERNAME}&password=${SSO_PASSWORD}&grant_type=password&client_id=admin-cli" ${sso_service}/realms/${SSO_REALM}/protocol/openid-connect/token`
+    token=`$CURL --data "username=${SSO_USERNAME}&password=${SSO_PASSWORD}&grant_type=password&client_id=admin-cli" ${sso_service}/realms/master/protocol/openid-connect/token`
```

Comment:
Why did this change?  This would mean that all clients would need access to the master realm, even if they belonged to another realm.

Reply:
If the issue is that this doesn't work if `SSO_REALM` is not set, then it might be easier to set `SSO_REALM=${SSO_REALM:-master}` before configuration is started.

Reply:
@rcernich The admin-cli client_id only exists in the master realm. This doesn't mean that the client will be created in the master realm, this is for retrieving the token that will be use later to create the client.

Reply:
Right, but shouldn't the uid/pwd use correspond to an admin user in the specific realm?  If that's not the case, then I'm wondering if anybody has used this with a realm other than master.

Reply:
@rcernich apparently the only requirement is that the user $SSO_USERNAME exists on this realm to get the token. This token is only to create new clients which should be allowed to any user of the realm. I will do the change. 
Thanks for the comments.